### PR TITLE
virtio-devices: block: Make latency calculation overflow safe

### PR DESCRIPTION
**Pull request by rbradford**
Fixes: #5745

Signed-off-by: Rob Bradford <rbradford@rivosinc.com>

### CODE REVIEW

**Comment by rbradford** [comment]
@praveen-pk Do you want to check if this works for you?

**Comment by praveen-pk** [comment]
@rbradford  I confirmed that this commit fixes the #5745 issue.  Thanks for the quick follow up.

**Comment by rbradford** [comment]
@likebreath This PR has a dockerfile bump so would be good if you could look at it today.

**Review by likebreath — COMMENTED** [review]
While reviewing this PR, I spotted another related bug #5712.

 Even with this bug (an incorrect smaller initial value), I wonder how a subtract overflow was introduced by #5712. I believe the cumulative average formula guarantees the final average result is also positive, e.g. should have no substract overflow.

File: virtio-devices/src/block.rs [hunk 1]
```diff
@@ -309,9 +309,10 @@ impl BlockEpollHandler {
                         read_avg = if read_avg == u64::MAX {
                             latency
                         } else {
-                            read_avg
-                                + ((latency * LATENCY_SCALE) - read_avg)
-                                    / (read_ops_last + read_ops.0)
+                            read_avg.saturating_add_signed(
+                                ((latency as i64).saturating_mul(LATENCY_SCALE as i64))
+                                    .saturating_sub(read_avg as i64),
+                            ) / (read_ops_last + read_ops.0)
```
[the remark this answers is not in the document]
Comment (rbradford) [reply virtio-devices/src/block.rs]: I think I fixed it...

**Pull request closed without merging**